File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        x = 0
        sum_digits = 0
        length = 1

        while n != 0:
            x = (n % 10) * length + x

            if n % 10 != 0:
                length = length * 10

            sum_digits += n % 10

            n = n // 10

        return sum_digits * x

File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.rs
impl Solution {
    pub fn sum_and_multiply(n: i32) -> i64 {
        let mut n = n;
        let mut x: i64 = 0;
        let mut sum: i64 = 0;
        let mut len: i64 = 1;

        while n != 0 {
            let digit = (n % 10) as i64;
            x = digit * len + x;

            if digit != 0 {
                len *= 10;
            }

            sum += digit;
            n /= 10;
        }

        sum * x
    }
}